src/seqn/trans.lisp

```diff
@@ -0,0 +1,399 @@
+(in-package :geb.seqn.trans)
+
+(defun remove0-dom (morphism)
+  (remove 0 (dom morphism)))
+
+;; Add range-n function
+
+(defun range-constraints-dom (domain)
+  (cond ((zerop (length domain))   nil)
+        ((= (length domain) 1)    (list (geb.vampir:range-n
+                                         (vamp:make-constant :const (car domain))
+                                         (vamp:make-wire :var
+                                                         (intern (format nil "x~a" 1))))))
+        (t (cons (geb.vampir:range-n (vamp:make-constant :const (car (last domain)))
+                                (vamp:make-wire :var
+                                                (intern (format nil "x~a" (length domain)))))
+            (range-constraints-dom (butlast
+                                    domain))))))
+
```

Comment:
formatting note, just newline the cond answers at this point

```lisp
(defun range-constraints-dom (domain)
  (cond ((zerop (length domain))
         nil)
        ((= (length domain) 1)
         (list (geb.vampir:range-n
                (vamp:make-constant :const (car domain))
                (vamp:make-wire :var
                                (intern (format nil "x~a" 1))))))
        (t
         (cons (geb.vampir:range-n
                (vamp:make-constant :const (car (last domain)))
                (vamp:make-wire :var
                                (intern (format nil "x~a" (length domain)))))
               (range-constraints-dom (butlast domain))))))
```

much easier to read. See my comments on this to see what else can be changed. This does not include the proper code changes.